app/models/reaction.rb

```diff
@@ -158,9 +161,19 @@ def viewable_by
 
   def assign_points
     base_points = BASE_POINTS.fetch(category, 1.0)
+
+    # Ajust for certain states
     base_points = 0 if status == "invalid"
     base_points /= 2 if reactable_type == "User"
     base_points *= 2 if status == "confirmed"
+
+    unless persisted? # Actions we only want to apply upon initial creation
+      # Author's comment reaction counts for more weight on to their own posts. (5.0 vs 1.0)
+      base_points *= 5 if positive_reaction_to_comment_on_own_article?
+
+      # New users will have their reaction weight gradually ramp by 0.1 from 0 to 1.0.
+      base_points *= new_user_adjusted_points if new_untrusted_user # New users get minimal reaction weight
+    end
```

Comment:
I thought reaction is created once and never touched again kind of object? other than that, wouldn't this be better in an `after_create` ?

Reply:
Reactions have a `status` attribute which can be updated after they are initially saved. This would be the one time they are touched after the fact. Those lines are reflected above in this method.

`status` is used for `pending` and `confirmed` vomit reactions. That's why this is `after_save`.

To me it seems like there could be more room for error if we try to break this up into multiple distinct callbacks vs conditional within one callback.